### ISSUE TITLE
Is there any autoencoder supported?

### ISSUE THREAD

**Issue by johnny5550822** [vlfeat/matconvnet#109]
Hi, I wonder if there is an implementation of autoencoder?

**Comment by vedaldi** [comment]
Hi, not yet, but it would be a very nice contribution :-)

**Comment by xcyan** [comment]
Any plans to incorporate autoencoder in official matconvnet? If not, our
team can contribute to that.

**Comment by vedaldi** [comment]
Hi,  our team is *not* working on an auto encoder just now (although we may have some related projects). We would be interested in an example of training an autoencoder network. I am thinking to create a contrib/ directory where some of these contributions could be initially added, and successively integrated in the core library when appropriate.

**Comment by johnny5550822** [comment]
I think it's a great idea to have a contrib/ because not all the work can be integrated right away to the core library

**Comment by umuguc** [comment]
I have a quick and dirty MNIST autoencoder implementation that is similar to https://github.com/BVLC/caffe/pull/330. You can find it here: https://github.com/umuguc/matconvnet/tree/master/examples/mnsitAutoencoder.

Here are some results after 108 epochs:

- http://i.imgur.com/h2WxSxZ.png (columns 2 and 4 are reconstructions of columns 1 and 2)
- validation: epoch 108: processing batch 100 of 100 ... 0.02 s (5366.0 images/s) err 87.4 (err is Euclidean loss)

It seems to be working fine, but there are no guarantees! ;-)

Note that it is not documented at all, but it should be straightforward to use. I can also push the pretrained model if anyone is interested.

**Comment by chandranikhil** [comment]
Hello umuguc, I would be interested in your pre-trained mnist model. Please push it.

**Comment by mingtop** [comment]
You did a really good job,man! @umuguc 
Thank u for you contribution on  euclibeanloss !
